SCF checkpointing update (#2190)
psi4/psi4: doc/sphinxman/source/external.rst

    psi4 -m

The mechanism for restarting HF/DFT calculations is described in details :ref:`here <scf:restart>`.

.. _`sec:save_wfn`:

Saving the wavefunction
=======================

A core object of |PSIfour| is the Wavefunction (short ``wfn``) object ::

  energy, wfn = energy('scf',return_wfn=True)

This C++/Python object (:py:class:`psi4.core.Wavefunction`) contains orbital
data, basis set information, result variables and more.
It can be saved either to a numpy file or converted to a python dictionary ::

  # write the wavefunction to file
  wfn.to_file('my_wfn')

  # alternatively store the dict representation of the wavefunction in memory
  wfn_dict = wfn.to_file()

> In either forms the content can be manipulated.

Reviewer: ```suggestion
In either form, its attributes can be set and edited.
```

The back conversion to a Wavefunction object uses the ``.from_file()`` functionality ::

  # read wavefunction from file

>   wfn_new = psi4.core.Wavefunction.from_file('my_wfn')

Reviewer 2: `wfn_new` is not the best name for an old wfn that has been loaded (as opposed to a new one made from scratch). How about `wfn_from_file` and `wfn_from_dict` below?

Author: It was taken directly from the test case, but you are right that it could get a better name.

  # make a wavefunction from the dict

>   wfn_from_dict = psi4.core.Wavefunction.from_file(wfn_dict)

Reviewer: `from_file`? Are you sure?

Author: For the `dict`? Yes, `from_file` can take both and there is no special `from` function for dicts.
